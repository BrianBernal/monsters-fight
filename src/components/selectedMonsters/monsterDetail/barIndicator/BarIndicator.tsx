// styles
import "./barIndicator.scss";

type TBarIndicator = {
  title: string;
  percentageVale: number;
};
function BarIndicator({ title, percentageVale }: TBarIndicator) {
  return (
    <div className="indicator-container">
      <span>{title}</span>
      <span className="indicator-container__bar">
        <span
          className="indicator-container__bar__filler"
          style={{ width: `${percentageVale}%` }}
        />
      </span>
    </div>
  );
}

export default BarIndicator;
