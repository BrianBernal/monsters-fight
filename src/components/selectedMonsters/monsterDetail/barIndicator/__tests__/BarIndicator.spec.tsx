// libraries
import renderer from "react-test-renderer";
import BarIndicator from "../BarIndicator";

describe("<BarIndicator />", () => {
  test("Renders correctly", () => {
    const title = "example title";
    const percentageVale = 90;
    const tree = renderer
      .create(<BarIndicator title={title} percentageVale={percentageVale} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
