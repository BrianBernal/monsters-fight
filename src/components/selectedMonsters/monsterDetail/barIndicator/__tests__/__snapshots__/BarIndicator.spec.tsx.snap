// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BarIndicator /> > Renders correctly 1`] = `
<div
  className="indicator-container"
>
  <span>
    example title
  </span>
  <span
    className="indicator-container__bar"
  >
    <span
      className="indicator-container__bar__filler"
      style={
        {
          "width": "90%",
        }
      }
    />
  </span>
</div>
`;
